File: frontend/src/types/dto/entity/game/GameDto.ts
import type { GameStatus } from '@/types/enum/GameStatus.ts';
import type { SharedGameStateDto } from './SharedGameStateDto';
import type { GamePlayerDto } from './GamePlayerDto';
import type { GameSettingsDto } from './GameSettingsDto';

export type GameDto = {
    gameId: number,
    lobbyId: number,
    gamePlayers: GamePlayerDto[],
    sharedGameState: SharedGameStateDto,
    initialBoardState: string,
    gameSettings: GameSettingsDto,
    gameStatus: GameStatus,
    gameStartsAt: string | null,
    gameEndsAt: string | null,
    endedPrematurely: boolean,
    gameEndedAt: string | null
}

File: backend/src/main/java/com/github/ryand6/sudokuweb/dto/entity/game/GameDto.java
package com.github.ryand6.sudokuweb.dto.entity.game;

import com.github.ryand6.sudokuweb.enums.Difficulty;
import com.github.ryand6.sudokuweb.enums.GameMode;
import com.github.ryand6.sudokuweb.enums.GameStatus;
import com.github.ryand6.sudokuweb.enums.TimeLimitPreset;
import lombok.*;

import java.time.Instant;
import java.util.Set;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class GameDto {

    private Long id;

    private Long lobbyId;

    private Set<GamePlayerDto> gamePlayers;

    private String initialBoardState;

    private SharedGameStateDto sharedGameState;

    private GameMode gameMode;

    private Difficulty difficulty;

    private TimeLimitPreset timeLimit;

    private GameStatus gameStatus;

    private Instant gameStartsAt;

    private Instant gameEndsAt;

}
